leaf-ui/js/object/Class.js

```diff
@@ -332,17 +332,30 @@ class AnalysisResult {
      *   (for nodeID 0000, time point 0, its satisfaction value is none)
      */
 
-    constructor() {
-        this.assignedEpoch;
-        this.timePointPath ;
-        this.timePointPathSize;
-        this.elementList; 
-        this.allSolution;
-        this.elementListPercentEvals;
-        this.isPathSim = false; //used for slider visualization
-        this.colorVis; //color visualization for analysis mode  
-        this.selectedTimePoint; //find where slider is initialized and set timepoint in here. Also place it in update function
-        this.maxTimePoint;
+    constructor(analysisResult) {
```

Comment:
Could you add a comment or something explaining why you made this change? Or just respond to this comment since I've forgotten lol

Reply:
Added a comment! We needed a way to make analysisResult objects that were loaded from JSON into actual instances of the AnalysisResult class

Reply:
Smart thank you! Sorry for my confusion